django_tidb/fields/vector.py

```diff
@@ -0,0 +1,144 @@
+import numpy as np
+from django.core import checks
+from django import forms
+from django.db.models import Field, FloatField, Func, Value
+
+MAX_DIM_LENGTH = 16000
+MIN_DIM_LENGTH = 1
+
+
+def encode_vector(value, dim=None):
+    if value is None:
+        return value
+
+    if isinstance(value, np.ndarray):
+        if value.ndim != 1:
+            raise ValueError("expected ndim to be 1")
+
+        if not np.issubdtype(value.dtype, np.integer) and not np.issubdtype(
+            value.dtype, np.floating
+        ):
+            raise ValueError("dtype must be numeric")
+
+        value = value.tolist()
+
+    if dim is not None and len(value) != dim:
+        raise ValueError("expected %d dimensions, not %d" % (dim, len(value)))
+
+    return "[" + ",".join([str(float(v)) for v in value]) + "]"
+
+
+def decode_vector(value):
+    if value is None or isinstance(value, np.ndarray):
+        return value
+
+    if isinstance(value, bytes):
+        value = value.decode("utf-8")
+
+    return np.array(value[1:-1].split(","), dtype=np.float32)
+
+
+class VectorField(Field):
```

Comment:
```suggestion
class VectorField(Field):
"""
Support for AI Vector storage.

Status: Beta

Info: https://www.pingcap.com/blog/integrating-vector-search-into-tidb-for-ai-applications/
"""
```

Maybe copy some more of the README into code comments and extend them